src/components/stepper-item/stepper-item.tsx

```diff
@@ -219,11 +213,7 @@ export class StepperItem implements InteractiveComponent, LocalizedComponent, Lo
             }
           >
             {this.icon ? this.renderIcon() : null}
-            {this.numbered ? (
-              <div class="stepper-item-number">
-                {numberStringFormatter.numberFormatter.format(this.itemPosition + 1)}.
-              </div>
-            ) : null}
+            {this.numbered ? <div class="stepper-item-number">{this.renderNumbers()}.</div> : null}
```

Comment:
We might want to move the numberStringFormatter check to the render block for all of its components to prevent similar bugs. It won't recreate a new NumberFormatter if the lang/numberingSystem is the same so putting the check in the render block won't be too expensive performance-wise. 

I can do that in a follow up PR